java/vortex-jni/build.gradle.kts

```diff
@@ -113,26 +142,65 @@ val platformLibSuffix =
     }
 
 val targetDir = projectDir.parentFile.parentFile.resolve("target")
+// TODO(aduffy): fetch the target triple dynamically
 val libraryFile = targetDir.resolve("release/libvortex_jni.$platformLibSuffix")
 
 val cargoCheck by tasks.registering(Exec::class) {
     workingDir = vortexJNI
     commandLine("cargo", "check")
 }
 
+val getTargetTriple by tasks.registering(Exec::class) {
+    workingDir = vortexJNI
+    commandLine("rustc", "--print", "target-triple")
+}
+
 val cargoBuild by tasks.registering(Exec::class) {
     workingDir = vortexJNI
-    commandLine(
-        "cargo",
-        "build",
-        "--release",
-    )
 
-    outputs.files(libraryFile)
+    val buildWithAsan = project.findProperty("buildWithAsan")?.toString()?.toBoolean() ?: false
+    println("buildWithAsan: $buildWithAsan")
+    if (buildWithAsan) {
+        // Force a rebuild
+        outputs.upToDateWhen { false }
+
+        // Get the target triple for the current platform. We need it
+        // so we can tell cargo to recompile the std crate for this target with ASAN enabled.
+        val output = ByteArrayOutputStream()
+        exec {
+            commandLine("rustc", "--print", "host-tuple")
+            standardOutput = output
+        }
+        val targetTriple = output.toString().trim()
+
+        // Build with ASAN to detect memory leaks and out of bounds accesses from Java.
+        environment("RUSTFLAGS", "-Zsanitizer=address")
+        commandLine(
+            "cargo",
+            "build",
+            "-Zbuild-std",
+            "--target",
+            targetTriple,
+            "--release",
+        )
+        // cargo puts the built artifact in a target-dependent directory when you specify the triple
+        outputs.files(targetDir.resolve("$targetTriple/release/libvortex_jni.$platformLibSuffix"))
+    } else {
+        commandLine(
+            "cargo",
+            "build",
+            "--release",
+        )
+        outputs.files(targetDir.resolve("release/libvortex_jni.$platformLibSuffix"))
+    }
```

Comment:
This allows us to optionally build the native library with llvm AddressSanitizer to catch memory leaks and invalid references. Pass the `-PbuildWithAsan=true` flag to gradle at build time